jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -463,6 +464,21 @@ public Runnable onFailure(Throwable x)
         return task;
     }
 
+    @Override
+    public Runnable onClose()
+    {
+        try (AutoLock ignored = _lock.lock())
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("onClose {} stream={}", this, _stream);
+
+            // If the channel doesn't have a stream, then no action is needed.
+            if (_stream == null)
```

Comment:
would `_request == null` be a better test.    Do we want an EofException if a connection is opened then closed without a request?

Reply:
I used the `_stream == null` test for consistency with `onFailure()` that tests for that too, but in practice it's identical to `_request == null` as both are reset to null at the same place in `recycle()`.

And I do believe it is correct to fail with a `EofException` when a connection is opened then immediately closed without sending a byte.

@sbordet Do you agree?